src/communication/Publisher.cpp

```diff
@@ -10,3 +10,32 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/communication/Publisher.h"
+#include <up-cpp/datamodel/builder/UMessage.h>
+
+namespace uprotocol::communication {
+using namespace uprotocol::datamodel::builder;
+
+Publisher::Publisher(std::shared_ptr<transport::UTransport> transport,
+          v1::UUri& topic, v1::UPayloadFormat format,
+          std::optional<v1::UPriority> priority,
+          std::optional<std::chrono::milliseconds> ttl) :
+    transport_(std::move(transport)) {
+
+    auto messageBuilder = UMessageBuilder::publish(std::move(topic))
+        .withPayloadFormat(format)
```

Comment:
If the builder becomes a regular member instead of a pointer, this could just be
```suggestion
    payload_builder_.withPayloadFormat(format)
```

Reply:
done